### ISSUE TITLE
multiple instances on one page

### ISSUE THREAD

**Issue by notslang** [NV/flying-focus#7]
If I have the flying focus chrome plugin installed and I visit a site that uses the flying focus script, 2 instances of flying focus run on the same page. This creates 2 separate elements (both with the id `flying-focus`) and they are both animated.

The fact that this happens doesn't actually break anything (they just overlap), and it's not really a performance concern since it's such a light plugin. Rather, it simply _feels_ wrong to have 2 of them running at the same time, especially since both of the elements created have the same id.

So, flying-focus should check if the `#flying-focus` element has already been created, and exit if it has, to prevent multiple instances.

**Comment by NV** [comment]
The extension used to check `#flying-focus`: [flying-focus.jspp.js#L2](https://github.com/NV/flying-focus/blob/6493a33ac8c32e43e1593de9b3f7f90069a0d492/standalone/flying-focus.jspp.js#L2). I removed when I implemented lazy initialization [flying-focus.js#L32-L35](https://github.com/NV/flying-focus/blob/1840dedc2a293d6992e23551514e773a31cc378c/chrome/flying-focus.js#L32-L35), e.g. the flying-focus element gets created on first focus event. In this situation race conditions happen.

I haven’t seen any websites besides mine using the script so I don’t think it isn’t very important issue as for now.

**Comment by notslang** [comment]
JS is inherently single-threaded, so one of the instances is bound to create the element first, right? Eh, anyway, you're right - it isn't a big issue.

**Comment by NV** [comment]
The problem is that flying-focus.js can be initialized either before or after the extension, depending on where script element is (head or end of the body).

**Comment by notslang** [comment]
exactly, so whichever happens first gets to continue running, the other
just stops. it doesn't matter whether the extension or script is the one
that is running, it's just whichever is first.

**Comment by nwinter** [comment]
[CodeCombat](http://codecombat.com/) is using it right now, and I see the issue. (I don't mind it at all; just adding an example of a site that uses it.)

**Comment by NV** [comment]
@slang800, that would work. I don’t have much time to test but I’ll accept a pull request.

@nwinter nice, I’m a fan! I just listed it on newly created [In the Wild](https://github.com/NV/flying-focus/wiki/In-the-Wild) wiki page.
